_includes/cloudcode/cloud-code.md

```diff
@@ -385,6 +385,27 @@ Parse.Cloud.beforeFind('MyObject2', (req) => {
 
 ```
 
+# beforeLogin Triggers
+
+*Available only on parse-server cloud code starting <version-number>*
+
+Sometimes you may want to run custom validation on a login request. The `beforeLogin` trigger can be used for blocking an account from logging in (for example, if they are banned), recording a login event for analytics, notifying user by email if a login occurred at an unusual IP address and more.
+
+```JavaScript
+Parse.Cloud.beforeLogin(async request => { 
+  const { object: user }  = request;
+  if(user.get('isBanned')) {
+   throw new Error('Access denied, you have been banned.')
+  }
+});
+```
+
+### Some considerations to be aware of:
+- It does not set the user on the request object - technically, the user has not yet been provided a session until after beforeLogin is successfully completed
+- It does not run on sign up
+- It does run on username/pass AND authProvider logins
+- It will not run if the login credentials are incorrect
```

Comment:
Extremely minor point, but it'll help with readability; could you list the positive scenarios first, followed by the negatives, so the user can build a better mental model in their head, i.e. Points 1 and 2 are scenarios where `beforeLogin` will run, Points 4 and 5 are when it will not.

Reply:
It's also worth adding that the login will wait for any asynchronous tasks to complete within the `beforeLogin` hook.

Reply:
Done